### DISCUSSION TITLE
Improve Accuracy of CO2 Sensor Automatic Self Calibration

### DISCUSSION THREAD

**Discussion by keenanjohnson** [Ribbit-Network/ribbit-network-frog-hardware#33] [General]
In #23 we discussed the tradeoffs in calibration methods for the CO2 sensor. We decided to pursue the ASC calibration method, but there will still be some inaccuracies introduced by using this method.

Let's keep the discussion going on strategies to mitigate this @eren-rudy and @daveb377.

**Comment by Derek-Gutheil** [comment]
Hey, I was looking into sensor accuracy today. I stumbled upon co2meter.com and after browsing their site for some time, sent them a question:

"
Hello!

My name is Derek Gutheil, and I am working on an open source project (https://www.ribbitnetwork.org/) with the goal of creating an global dataset of outdoor CO2 levels.

Seeing as we are measuring CO2 levels in the 400 to 800 ppm range, and I hoping you can give us some advice on how to accurately measure this low level of CO2. I have looked into your sensors and it seems like the most accurate sensors that you have achieve +-30ppm and 3% MV. As you show in your example (https://www.co2meter.com/blogs/news/how-to-measure-carbon-dioxide), this gets an effective range of 358ppm to 442ppm for a 400ppm sample.

So I have a few questions:
1. Are there other sensors in development or on the market that you know of that can get down to +-5-10ppm accuracy?
2. Are there industry standard techniques to get +-5-10ppm accuracy out of +-30ppm sensors?
3. Finally, are there more expensive sensors with higher accuracy that we can use to calibrate the sensors in our devices?

Thanks so much for the help,
-Derek Gutheil
"

Overall, the lack of accuracy and precision on these sensors is definitely concerning. Have there been any updates on your side recently?

**Reply by keenanjohnson** [reply]
Hey @Derek-Gutheil! Great job reaching out! I haven't heard any new updates and I'm not aware of any sensors at a relatively inexpensive pricepoint with a higher accuracy, unfortunately. I think this is an area that deserves continuous focus and development though!

**Reply by Derek-Gutheil** [reply]
They just responded and linked me to one of their sensors, so no luck going down that road unfortunately.

I am definitely concerned about this and I think we need a real plan to handle calibration in the field. The only thing worse than no data is bad data.

**Reply by keenanjohnson** [reply]
Indeed! I share the concern!

**Reply by leckerschmecker38** [reply]
HI  all .. i am  professionally involved in    gas sensing  ..  i   look into this project here  for   a  littel bit  and  wonder   if anyone    has done the  research to actually determine   the   sensor specifications  needed   for  the purpose of the project  .. 

i understood  this here is  about  a  global CO2  sensor  network ,,.. 

but  what do you want to see / detect / qauantify .. 
A) global  variations in    background Co2  due to  industry   ..   e.g  combustion 
B)  local     variations in   urban  /  slightly  polluted  areas  ( Cities  in   developed  countries ) 
C)   local variations   of  co2  near   indstrial areas   or cities in  developping  countries   with proably  strongly  polluted air 

case  a ) you  will  need  to detect  around   400 ppm   ..   but the  global average CO2  rises currently  with  2,5 ppm  PER YEAR !  you will  need   >>  the   sensors  with  best stability and the best    accuracy   ..  i would  guess you will face  quite  some problem  if   you try this with a  sensor  with   a    resolution of  30  ppm   .. not sure  what the stabiliyt of the  SCD30 is  over  a  year 

case B)   CO2  will be  between  400  and    800  ..     .. there will be daily   ups and downs   as soon as  you get   some how near traffic  .. at  night  co2   will  be domintaed by other  sources    an  during  rush hours  you will find big  peaks  ...   so depending on    where   you place your sensor   , what  fetch  it  has  and      near  which  emmiters  you ar e you  will get  highly  variable   Co2  fluctuations an    those  will be    mixed and modified by transport  ( caused  by  wind ,  temperature , pressure and humidiuty ) 
so here you  will have  less problems with  30 ppm resolution  .. but  you  will   face an interpreatation problem  if  your co2  fluctuationsa  are caused by  the senosr   or   by  your  sources  and   especially by    which  sources 


case  c  )  polluted  cities   even  higher  co2  values and higher   dynamics ,,   but  how  do you want to do the source attribution  ?  


human disturbances ,,   
everone using the  senosr   should be made aware  that  your breath  contains up to   50.000  ppm of  co2 .. so   if the  sensor   get   contact with any  of your  exhaled air  or    from someone   nearby  you  can also  cause  very big  fluctioations    and  even saturation 

other disturbances 
the  Co2  NDIR  sensoor uses  light   and  some  sort of    light filter   to become specific .. 
if there is any other  gas absorbing  in  the  selected  light   you will get   cross dependencies ., 
in particular in   urban  outdoor areas  , near industry   etc   - particularly in  developping  countries  with   bad  airquality  you have to  be aware of   cross sensitivities  of your sensor  to other    gas species ,,  like  hydrocarbons  , VOCs  etc  pp 

back to the  sensor  and sensor system  

for  case  A) i  you need  a  stability and accuracy of  about 1 ppm CO2    to detect and annual  change of  2.5  ppm ..   for this i think you need  a lab based  study  to detemine  the perfromacne  limits  of your sensor  ( alone) and as the  sensor systems ( incl your electronics)   sofar there is  a factor of  10  or more missing   to    be a solution for case a) .. and there  needs  to be  a  valdiation strategy   to  proove  this performance  and   you need  some sort of   reference  to check you sensors  .. you  will  also need  a    site  selection and valiidation stategy  ..   in order to    select a   site suitable for  background  Co2 determination 

case B)  and  c)   
yoiu still need  a applicaiton  derived   definition of the  required  sensor performance 
and you  need   a  method   to  quantify  your  sensor performance 

also  you  need additional infomration   to   provide some informaiton   on the  transport problem  ( where does  your   gas  come from  which  goes in the sensor) and you  should better have  some infomration on   the  dominating  sources     which are in  the fetch   of  your sensor   .. those  things  will  take  a  lot of  work  to   inquire 

if you dont have  this infomration   you  will    to  put it  bluntly  just have  a  fluctuation  sensor  signal   .. but  what  infomration do you want to derive  from  this 


i am  aware that  my    comments  here   might be a  bit  overwhleming or  even   spoiling  the  story ..   
but    it takes  a  lot of  research and background knowledge  (  on  environment , sensor , transport processes  etc etc ) 
to  design   - 
setup - 
contruct - 
operate - 
read out   
validate 
a   sensor network  and its elements 
and in particular   
to interpret   the  signals and  output 

i am  also  fascinated  by  the  low cost  , size   weight and  power requirment  of  these intergrated sensors 
but 
the  big question   is  .. how can the    ambient / environmental  signals  / patterns /  sources  / sinks  and  gas transport   so you are  looking   really  for be extracted  form  the  lower  qualit signals  of the  lower cost  sensors   ?  or   is  the infomration   so disturbed by  the noise and  drift  and   problems   which   are  caused by the   lower  costs   that  they  cant  be  extraxted 

and  another  question is  .,   can  you answer the above  question    without  emplyoing some  sort of  reference  sensor ( of  much higher performance and much higher  cost )   ? 


looking forward  to  discuss  this  

and  if  i find enough  time  i   will try  to  provide  some  answer  or  at least   informaitions  to help  you in  tackling this  questions  

best    regards 

leckerschmecker38
